service/worker/pernamespaceworker.go

```diff
@@ -377,12 +383,21 @@ func (w *perNamespaceWorker) handleError(err error) {
 		w.logger.Error("bug: handleError found existing timer")
 		return
 	}
-	sleep := w.retrier.NextBackOff()
-	if sleep < 0 {
-		w.logger.Error("Failed to start sdk worker, out of retries", tag.Error(err))
-		return
+
+	var sleep time.Duration
+
+	if retryAfter, ok := err.(errRetryAfter); ok {
+		// asked for an explicit delay due to rate limit, use that
+		sleep = time.Duration(retryAfter)
```

Comment:
I was initially wondering if it would improve the observability by adding an info log, but came to the conclusion it's probably too noisy.

Reply:
yeah, consider the situation where we have 100 namespaces and it takes 10s to start the workers (by default). it'll just log the starts slowly, it should be pretty clear to see what's happening.